crates/bls-crypto/src/bls/signature.rs

```diff
@@ -66,6 +66,24 @@ impl Signature {
             .into()
     }
 
+    /// Computes the dot product of a vector of signatures and a vector of exponents.
+    pub fn batch(r: &[Fr], signatures: &[&Signature]) -> Signature {
```

Comment:
Arkworks doesn't verify that both vecs have the same length, but takes the smaller: https://github.com/arkworks-rs/algebra/blob/e7eeea3f2ac7d621f68089b978097403dd02e91b/ec/src/msm/variable_base.rs#L16

Do you think we need to check it on our side?